web/data/accounts.go

```diff
@@ -167,3 +174,45 @@ func createAccount(c echo.Context) error {
 		"data": doc.ToMapWithType(),
 	})
 }
+
+// CozyMetadataFromClaims returns a CozyMetadata struct, with the app fields
+// filled with information from the permission claims.
+func CozyMetadataFromClaims(c echo.Context) *metadata.CozyMetadata {
+	cm := metadata.New()
+
+	var slug, version string
+	if claims := c.Get("claims"); claims != nil {
+		cl := claims.(permission.Claims)
+		switch cl.Audience {
+		case consts.AppAudience, consts.KonnectorAudience:
+			slug = cl.Subject
+		case consts.AccessTokenAudience:
+			if perms, err := middlewares.GetPermission(c); err == nil {
+				if cli, ok := perms.Client.(*oauth.Client); ok {
+					slug = oauth.GetLinkedAppSlug(cli.SoftwareID)
+					// Special case for cozy-desktop: it is an OAuth app not linked
```

Comment:
Well, I don't think we need this special case for cozy-desktop, as this app doesn't create any io.cozy.accounts

Reply:
It doesn't at the moment but I thought it would be nice to handle this special case right away. I fear we might not think about this part if we start creating accounts from Desktop some day.